src/Microsoft.Dnx.TestHost/ProjectCommand.cs

```diff
@@ -62,10 +61,76 @@ private static string GetVariable(IApplicationEnvironment environment, string ke
             return Environment.GetEnvironmentVariable(key);
         }
 
-        private static Task<int> ExecuteMain(IServiceProvider services, string entryPoint, string[] args)
+        private static Task<int> ExecuteMain(string entryPointName, TestHostServices services, string[] args)
+        {
+            var assembly = Assembly.Load(new AssemblyName(entryPointName));
+            object instance;
+            MethodInfo entryPoint;
+            if (TryGetEntryPoint(assembly, instance: out instance,  entryPoint: out entryPoint))
+            {
+                var result = entryPoint.Invoke(instance, new object[] { services, args });
+                if (result is int)
+                {
+                    return Task.FromResult((int) result);
+                }
+                return Task.FromResult(0);
+            }
+
+            return EntryPointExecutor.Execute(assembly, args, null);
+        }
+
+        //Copied from EntryPointExecutor because it would be gone
+        public static bool TryGetEntryPoint(Assembly assembly, out object instance, out MethodInfo entryPoint)
+        {
+            string name = assembly.GetName().Name;
+
+            instance = null;
+
+            // Add support for console apps
+            // This allows us to boot any existing console application
+            // under the runtime
+            entryPoint = GetEntryPoint(assembly);
+            if (entryPoint != null)
+            {
+                return true;
+            }
+
+            var programType = assembly.GetType("Program") ?? assembly.GetType(name + ".Program");
+
+            if (programType == null)
+            {
+                var programTypeInfo = assembly.DefinedTypes.FirstOrDefault(t => t.Name == "Program");
+
+                if (programTypeInfo == null)
+                {
+                    System.Console.WriteLine("'{0}' does not contain a static 'TestMain' method suitable for an entry point", name);
```

Comment:
Just realized this message is complaining about `TestMain()` when the code can't even find the `Program` class. Wonky.